floss/main.py

```diff
@@ -1062,153 +599,152 @@ def main(argv=None):
     :param argv: optional command line arguments, like sys.argv[1:]
     :return: 0 on success, non-zero on failure
     """
-    logging.basicConfig(level=logging.WARNING)
-
-    parser = make_parser()
-    if argv is not None:
-        options, args = parser.parse_args(argv[1:])
-    else:
-        options, args = parser.parse_args()
-
-    set_log_config(options.debug, options.verbose)
+    parser = make_parser(argv or sys.argv[1:])
+    args = parser.parse_args(args=argv)
 
-    sample_file_path = parse_sample_file_path(parser, args)
-    min_length = parse_min_length_option(options.min_length)
+    set_log_config(args)
 
     # expert profile settings
-    if options.expert:
-        options.save_workspace = True
-        options.group_functions = True
-        options.quiet = False
+    # TODO: removeme
+    args.expert = args.x
+    args.should_show_metainfo = True
+    args.save_workspace = True
+    args.group_functions = True
+    args.quiet = False
+
+    # set defaults when -x is not provided
+    args.min_length = args.min_length if hasattr(args, "min_length") else DEFAULT_MIN_LENGTH
+    args.functions = args.functions if hasattr(args, "functions") else None
+    args.no_filter = args.no_filter if hasattr(args, "no_filter") else False
+    args.max_instruction_count = (
+        args.max_instruction_count if hasattr(args, "max_instruction_count") else DEFAULT_MAX_INSN_COUNT
+    )
+    args.max_address_revisits = (
+        args.max_address_revisits if hasattr(args, "max_address_revisits") else DEFAULT_MAX_ADDRESS_REVISITS
+    )
+    args.no_static_strings = args.no_static_strings if hasattr(args, "no_static_strings") else False
+    args.no_decoded_strings = args.no_decoded_strings if hasattr(args, "no_decoded_strings") else False
+    args.no_stack_strings = args.no_stack_strings if hasattr(args, "no_stack_strings") else False
+    args.is_shellcode = args.is_shellcode if hasattr(args, "is_shellcode") else False
+    args.shellcode_entry_point = args.shellcode_entry_point if hasattr(args, "shellcode_entry_point") else None
+    args.shellcode_base = args.shellcode_base if hasattr(args, "shellcode_base") else None
```

Comment:
i'll address this in a subsequent PR that refactors the output rendering (and main() extraction logic).